pkg/run/run.go

```diff
@@ -175,38 +176,81 @@ func (r *RunOptions) PreRunCheck(kclient kubernetes.Interface) error {
 
 	// sonobuoy namespace exists so return error
 	if p.Name != "" {
-		return errors.New("sonobuoy namespace already exists")
+		return errors.New(fmt.Sprintf("%s namespace already exists", pkg.CertificationNamespace))
 	}
 
 	if r.dedicated {
+
 		log.Info("Ensuring proper node label for dedicated mode")
 		nodes, err := coreClient.Nodes().List(context.TODO(), metav1.ListOptions{
-			LabelSelector: "node-role.kubernetes.io/tests=",
+			LabelSelector: pkg.DedicatedNodeRoleLabelSelector,
 		})
 		if err != nil {
 			return err
 		}
 		if nodes.Items != nil && len(nodes.Items) == 0 {
 			return errors.New("No nodes with role required for dedicated mode (node-role.kubernetes.io/tests)")
 		}
+
+		// Skip preflight checks and create namespace manually with Tolerations
+		tolerations, err := json.Marshal([]v1.Toleration{{
+			Key:      pkg.DedicatedNodeRoleLabel,
+			Operator: v1.TolerationOpExists,
+			Value:    "",
+			Effect:   v1.TaintEffectNoSchedule,
+		}})
+		if err != nil {
+			return err
+		}
+
+		namespace = &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: pkg.CertificationNamespace,
+				Annotations: map[string]string{
+					"scheduler.alpha.kubernetes.io/defaultTolerations": string(tolerations),
+					"openshift.io/node-selector":                       pkg.DedicatedNodeRoleLabelSelector,
+				},
+			},
+		}
+	} else {
+		namespace = &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: pkg.CertificationNamespace,
+			},
+		}
+	}
+
+	_, err = kclient.CoreV1().Namespaces().Create(context.TODO(), namespace, metav1.CreateOptions{})
+	if err != nil {
+		return err
 	}
 
+	// Create Sonobuoy ServiceAccount
+	serviceAccount := &v1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pkg.SonobuoyServiceAccountName,
+			Namespace: pkg.CertificationNamespace,
+			Labels: map[string]string{
+				"component": pkg.SonobuoyComponentLabelValue,
+			},
+		},
+	}
+	_, err = kclient.CoreV1().ServiceAccounts(pkg.CertificationNamespace).Create(context.TODO(), serviceAccount, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
 	log.Info("Ensuring the tool will run in the privileged environment...")
+
 	// Configure SCC
 	anyuid := &rbacv1.ClusterRoleBinding{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: pkg.AnyUIDClusterRoleBinding,
+			Name: "system:openshift:scc:anyuid",
 		},
 		Subjects: []rbacv1.Subject{
 			{
-				Kind:     rbacv1.GroupKind,
+				Kind:     rbacv1.ServiceAccountKind,
 				APIGroup: rbacv1.GroupName,
```

Comment:
Since ServiceAccounts are core group, I think this should be `""` or just unspecified.